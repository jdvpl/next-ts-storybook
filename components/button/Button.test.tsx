import React from "react";
import { render, screen } from "@/test-utils";
import { Button } from "./Button";
import userEvent from "@testing-library/user-event";
describe("Button test cases", () => {
  it("render check", () => {
    const onClick = jest.fn();
    const { asFragment } = render(<Button onClick={onClick}>Buton</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
  it("Check onclick Callcall", async () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Buton</Button>);

    const element = screen.getByRole("button");
    await userEvent.click(element);

    expect(onClick).toHaveBeenCalled();
  });
});
