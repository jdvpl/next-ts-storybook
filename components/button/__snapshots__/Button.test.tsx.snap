// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases render check 1`] = `
<DocumentFragment>
  <button
    class="css-1trr18r"
    color="primary"
  >
    Buton
  </button>
</DocumentFragment>
`;
